### PR TITLE
TTBR support for ReceiveOnly on MSMQ

### PR DESCRIPTION
**Pull request by timbussmann**
Relates to https://github.com/Particular/NServiceBus/pull/4218

NSB v6.1 will support ttbr when using ReceiveOnly transaction mode.

ping @Particular/docs-maintainers @ramonsmits

### CODE REVIEW

File: nservicebus/messaging/discard-old-messages_msmq_core_[6.0,).md [hunk 1]
```diff
@@ -0,0 +1,3 @@
+MSMQ continuously checks the TimeToBeReceived attribute of all queued messages. As soon as the message has expired, it is removed from the queue, and disk space reclaimed. MSMQ will however only allow a single TimeToBeReceived for all messages in a transaction. It will silently copy the TimeToBeReceived from the first message enlisted to all other messages in the same transaction, leading to a potential message loss scenario. This issue, unfortunately, make it impossible to support setting TimeToBeReceived for messages sent from a transactional MSMQ endpoint.
+
+Note: Besides `TransportTransactionMode.None`, NServiceBus Version 6.1 and above also support the TimeToBeReceived setting for `TransportTransactionMode.ReceiveOnly`.
```
Comment (SzymonPobiega) [inline nservicebus/messaging/discard-old-messages_msmq_core_[6.0,).md]: 6.1 ?

File: nservicebus/messaging/discard-old-messages_msmq_core_[6.0,).md [hunk 2]
```diff
@@ -0,0 +1,3 @@
+MSMQ continuously checks the TimeToBeReceived attribute of all queued messages. As soon as the message has expired, it is removed from the queue, and disk space reclaimed. MSMQ will however only allow a single TimeToBeReceived for all messages in a transaction. It will silently copy the TimeToBeReceived from the first message enlisted to all other messages in the same transaction, leading to a potential message loss scenario. This issue, unfortunately, make it impossible to support setting TimeToBeReceived for messages sent from a transactional MSMQ endpoint.
```
Comment (SzymonPobiega) [inline nservicebus/messaging/discard-old-messages_msmq_core_[6.0,).md]: This para seems to be identical to the V5 one so why extracting it to a snippet?

File: nservicebus/messaging/discard-old-messages_msmq_core_[6.0,).md (hunk 1 above)
Comment (timbussmann) [inline nservicebus/messaging/discard-old-messages_msmq_core_[6.0,).md]: 6.1 !

File: nservicebus/messaging/discard-old-messages_msmq_core_[6.0,).md (hunk 2 above)
Comment (SzymonPobiega) [inline nservicebus/messaging/discard-old-messages_msmq_core_[6.0,).md]: 👍 for the note to be in partial but the first paragraph is the same for both so I think should stay in the main doco for now. No reason for duplicating it just now.

Comment (timbussmann) [inline nservicebus/messaging/discard-old-messages_msmq_core_[6.0,).md]: good with that, will change

**Pull request merged**